2022-08-30 20:25:58,591 ERROR: Exception on /explore [GET] [in C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask\app.py:1741]
Traceback (most recent call last):
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask\app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask\app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask\app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask\app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
  File "C:\Users\User\PycharmProjects\flask_miniblog\app\routes.py", line 131, in explore
    return render_template('index.html', title='Explore', posts=posts)
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask\templating.py", line 147, in render_template
    return _render(app, template, context)
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask\templating.py", line 130, in _render
    rv = template.render(context)
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\jinja2\environment.py", line 1301, in render
    self.environment.handle_exception()
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\User\PycharmProjects\flask_miniblog\app\templates\index.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "C:\Users\User\PycharmProjects\flask_miniblog\app\templates\base.html", line 33, in top-level template code
    {% block content %}{% endblock %}
  File "C:\Users\User\PycharmProjects\flask_miniblog\app\templates\index.html", line 18, in block 'content'
    {% for post in posts %}
TypeError: 'Pagination' object is not iterable
2022-08-30 20:26:00,539 ERROR: Exception on /index [GET] [in C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask\app.py:1741]
Traceback (most recent call last):
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask\app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask\app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask\app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask\app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
  File "C:\Users\User\PycharmProjects\flask_miniblog\app\routes.py", line 30, in index
    return render_template('index.html', title='Home page', form=form, posts=posts)
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask\templating.py", line 147, in render_template
    return _render(app, template, context)
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask\templating.py", line 130, in _render
    rv = template.render(context)
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\jinja2\environment.py", line 1301, in render
    self.environment.handle_exception()
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\User\PycharmProjects\flask_miniblog\app\templates\index.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "C:\Users\User\PycharmProjects\flask_miniblog\app\templates\base.html", line 33, in top-level template code
    {% block content %}{% endblock %}
  File "C:\Users\User\PycharmProjects\flask_miniblog\app\templates\index.html", line 18, in block 'content'
    {% for post in posts %}
TypeError: 'Pagination' object is not iterable
2022-08-30 20:30:31,111 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:42]
2022-08-30 20:43:49,262 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:42]
2022-08-30 20:51:33,665 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:42]
2022-08-31 21:01:21,839 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:44]
2022-08-31 21:03:19,968 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:44]
2022-09-01 19:54:06,260 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:46]
2022-09-01 19:57:10,573 ERROR: Exception on /reset_password_request [POST] [in C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask\app.py:1741]
Traceback (most recent call last):
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask\app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask\app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask\app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\User\PycharmProjects\flask_miniblog\env\lib\site-packages\flask\app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "C:\Users\User\PycharmProjects\flask_miniblog\app\routes.py", line 173, in reset_password_request
    send_password_reset_email(user)
  File "C:\Users\User\PycharmProjects\flask_miniblog\app\email.py", line 20, in send_password_reset_email
    token = user.get_reset_password_token()
  File "C:\Users\User\PycharmProjects\flask_miniblog\app\models.py", line 60, in get_reset_password_token
    return jwt.encode(
AttributeError: 'str' object has no attribute 'decode'
2022-09-01 20:30:11,889 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:48]
2022-09-01 21:21:30,077 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:52]
2022-09-01 21:28:15,453 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:52]
2022-09-04 17:33:50,455 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:52]
2022-09-04 17:35:06,453 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:52]
2022-09-04 17:37:37,388 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:52]
2022-09-04 17:48:17,635 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:52]
2022-09-04 17:48:51,768 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:52]
2022-09-04 17:49:09,412 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:52]
2022-09-04 18:20:02,149 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:52]
2022-09-04 18:39:05,059 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:74]
2022-09-04 18:41:45,448 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:74]
2022-09-04 20:44:07,281 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:74]
2022-09-04 20:53:09,148 INFO: Microblog startup [in C:\Users\User\PycharmProjects\flask_miniblog\app\__init__.py:74]
